src/model/floorplan.ts

```diff
@@ -213,35 +216,58 @@ module BP3D.Model {
           'backTexture': wall.backTexture
         });
       });
-      floorplan.newFloorTextures = this.floorTextures;
+      floorplan.newFloorTextures = <any> this.floorTextures;
       return floorplan;
     }
 
-    public loadFloorplan(floorplan) {
+    public loadFloorplan(floorplan: BP3D.IO.IFloorPlan) {
       this.reset();
 
+      var tThick = BP3D.Core.Configuration.getNumericValue(BP3D.Core.configWallThickness);
+      BP3D.Core.Configuration.setValue(BP3D.Core.configWallThickness, tThick * BP3D.Three.CmToWorld);
```

Comment:
Why do we override this config value with a scaled version of itself? Is there a way to avoid this?